### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by leewang0**
## Description of the change

- Move away from OpenAPI Generator's `typescript` generator in favor of `typescript-fetch`, which is a different set of templates that uses the js `fetch` api to make requests
- Change API classes to use request models instead of having individual params in method calls
- Instead of each category having duplicates of base api classes, pagination and remote_data we now share them
- Remove `Paginated___List` models in favor of using generics (`MergePaginatedResponse<T>`)
- Remove uses of `any` in favor of `JSONValue`, a type which matches the json language spec (albeit with `undefined` and `null` as possible values)
- add an example simple-test which calls an api from each category. Redact sensitive values.

Screen Shot 2022-07-29 at 2 21 17 PM

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests, or the description above explains how testing was performed

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or gif attached. Download "Kap" from the Mac App store for easy gif screen capture.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,40 +1,45 @@
-# Merge TypeScript SDK
+## @mergeapi/merge-sdk-typescript@2.0.0
 
-The SDK for interacting with Merge's unified Accounting, ATS, CRM, HRIS and Ticketing API's.
+This generator creates TypeScript/JavaScript client that utilizes [Fetch API](https://fetch.spec.whatwg.org/). The generated Node module can be used in the following environments:
```
Comment (eltonkl) [inline README.md]: `creates a`

File: src/remote_data.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Merge Accounting API
+ * The unified API for building rich integrations with multiple Accounting & Finance platforms.
```
Comment (eltonkl) [inline src/remote_data.ts]: probably wanna strip category from these shared files

File: src/merge_paginated_response.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Merge CRM API
+ * The unified API for building rich integrations with multiple CRM platforms.
```
Comment (eltonkl) [inline src/merge_paginated_response.ts]: also here

File: src/remote_data.ts (hunk 2 above)
Comment (leewang0) [reply src/remote_data.ts]: rip now you know which category i based the shared files on :(

**Review by eltonkl — APPROVED** [review]

**Pull request merged**